src/quo2/components/buttons/dynamic_button.cljs

```diff
@@ -0,0 +1,97 @@
+(ns quo2.components.buttons.dynamic-button
+  (:require [quo.theme :as theme]
+            [quo.react-native :as rn]
+            [reagent.core :as reagent]
+            [status-im.i18n.i18n :as i18n]
+            [quo2.components.icon :as icon]
+            [quo2.foundations.colors :as colors]
+            [quo2.components.markdown.text :as text]))
+
+(defn- in?
+  "true if coll contains elm"
+  [coll elm]
+  (some #(= elm %) coll))
+
+(defn- get-button-color [type pressed?]
+  (if (in? '(:jump-to :mention) type)
+    (if pressed?
+      (colors/theme-colors colors/primary-60 colors/primary-50)
+      (colors/theme-colors colors/primary-50 colors/primary-60))
+    (if pressed?
+      (colors/theme-colors colors/neutral-80-opa-80 colors/white-opa-80)
+      (colors/theme-colors colors/neutral-80-opa-70 colors/white-opa-70))))
+
+(defn- get-icon-and-text-color [type]
+  (if (in? '(:jump-to :mention) type)
+    colors/white
+    (colors/theme-colors colors/white colors/neutral-100)))
+
+(defn- icon-view [type]
+  [icon/icon
+   (case type
+     :jump-to      :main-icons2/jump-to
+     :mention      :main-icons2/mention
+     :notification :main-icons2/arrow-down
+     :search       :main-icons2/search
+     :search2      :main-icons2/search
+     :bottom       :main-icons2/arrow-down)
+   {:size            12
+    :color           (get-icon-and-text-color type)
+    :container-style {:margin-top    6
+                      :margin-bottom 6
+                      :margin-left   (case type
+                                       :jump-to      0
+                                       :mention      8
+                                       :notification 2
+                                       :search       8
+                                       :search2      6
+                                       :bottom       6)
+                      :margin-right  (case type
+                                       :jump-to      8
+                                       :mention      2
+                                       :notification 8
+                                       :search       4
+                                       :search2      6
+                                       :bottom       6)}}])
+
+(defn dynamic-button
+  "[:dynamic-button opts]
+   opts
+   {:type        :jump-to/:mention/:notification/:search/:search2/:bottom
```

Comment:
what's the difference between search and search2?

Reply:
https://discord.com/channels/624205794384281629/626744384565674034/1028970164437393480

Reply:
okay, consider changing the name to something more descriptive so. e.g :search-with-label